sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/ConnectionStringTokenParser.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Azure.Messaging.EventHubs.Tests
+{
+    /// <summary>
+    ///   General purpose parser that allows to retrieve a token in connection string.
+    /// </summary>
+    ///
+    internal static class ConnectionStringTokenParser
+    {
+        /// <summary>
+        ///   It parses a connection string looking for the provided token name and returns its value.
+        /// </summary>
+        ///
+        /// <param name="connectionString">An Event Hubs namespace connection string</param>
+        /// <param name="tokenKey">The token to be looked for</param>
+        ///
+        /// <returns>The value of the token if found; otherwise, a <see cref="InvalidOperationException" /> is thrown.</returns>
+        ///
+        internal static string ParseTokenAndReturnValue(string connectionString, string tokenKey)
+        {
+            string token = ParseToken(connectionString, tokenKey);
+
+            if (string.IsNullOrEmpty(token))
+            {
+                throw new InvalidOperationException($"The { tokenKey } token was not found.");
+            }
+
+            string tokenValue = ParseTokenValue(token);
+
+            if (string.IsNullOrEmpty(tokenValue))
+            {
+                throw new InvalidOperationException($"The { tokenKey } token is not well-formed.");
+            }
+
+            return tokenValue;
+        }
+
+        /// <summary>
+        ///   It parses a connection string looking for the provided token and returns it.
+        /// </summary>
+        ///
+        /// <param name="connectionString">An Event Hubs namespace connection string</param>
+        /// <param name="tokenKey">The token key to be looked for</param>
+        ///
+        /// <returns>The token if found; otherwise, null.</returns>
+        ///
+        internal static string ParseToken(string connectionString, string tokenKey)
+        {
+            var regex = new Regex($"(?<token>{ tokenKey }=[^;]+(;|$))", RegexOptions.IgnoreCase);
+
+            if (regex.IsMatch(connectionString))
+            {
+                return regex.Match(connectionString).Groups["token"].Value;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        ///   It parses a token and returns its value.
+        /// </summary>
+        ///
+        /// <param name="token">The token to be looked for in the form "Key=Value;"</param>
+        ///
+        /// <returns>The value of the token if found; otherwise, null.</returns>
+        ///
+        internal static string ParseTokenValue(string token)
+        {
+            var regex = new Regex("(?<tokenkey>[^;]+)=(?<tokenvalue>[^;]+)(;|$)", RegexOptions.IgnoreCase);
```

Comment:
nit:  You may want to consider hoisting the `RegEx`  instances that are static strings to `static readonly` members and set a regex option for `Compiled`.    Not a concern, given the test context, but may be helpful.